Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -150,18 +154,29 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
           foreach (var assertionNode in assertions) {
             if (assertionNode.Range.Contains(position) ||
                 assertionNode.ImmediatelyRelatedRanges.Any(range => range.Contains(position))) {
-              if (information != "") {
-                information += "\n\n";
-              }
 
-              information += GetAssertionInformation(state, position, assertionNode, assertionBatch,
-                assertionIndex, assertionBatchCount, node);
+              var whereToAdd = assertionNode.StatusVerification == GutterVerificationStatus.Error ?
+                errors : other;
+              var content = GetAssertionInformation(state, position, assertionNode, assertionBatch,
+                                            assertionIndex, assertionBatchCount, node);
+              var resources = assertionBatch.ResourceCount;
+              whereToAdd.Add((content, resources));
             }
 
             assertionIndex++;
           }
         }
 
+        var biggerResourceCountFirst =
+          Comparer<(string content, long resources)>.Create(
+          (left, right) =>
+            (int)(right.resources / 1000000) - (int)(left.resources / 1000000)
```

Comment:
Is there any need to do this division just to compare resource counts?

Reply:
Good question, the comparison metric requires an "int", when the resource count is a long. I have seen resource counts exceeding the range of an int in practice and display a negative result, but they should be displayed first.
I divided by one million to ensure we were practically within the int range, but I decided to do better. In the newest version, I'm mapping algorithmically long to int and added a comment for that.

Reply:
I think the only significance of the `int` result is whether it's less than, equal to, or greater than zero. So I think `right.resources.CompareTo(left.resources)` would work.

Reply:
Oh yeah did not know this method existed, and I could have implemented it myself. Thanks for the simplification.